lib/setup/setupInstall.js

```diff
@@ -1586,8 +1592,65 @@ function Install(options) {
                 // find the adapter's io-package.json
                 const adapterNpm = `${tools.appName}.${adapter}`;
                 const ioPack = require(`${adapterNpm}/io-package.json`); // yep, it's that easy
+
                 if (!ioPack.common || !ioPack.common.nondeletable) {
                     await that.npmUninstallAsync(adapterNpm, null, false);
+                    // after uninstalling we have to restart the defined adapters
+                    if (ioPack.common.restartAdapters) {
+                        if (typeof ioPack.common.restartAdapters !== 'object') {
```

Comment:
WOuld not be Array.isArray() as check here more specific? Object could also be {...} which would break when accessing [0]

Reply:
True, the logic is partially copied out of setupUpload, where it works like that.

Reply:
Then we shuld fix that pot in both places

Reply:
But I guess, it won't change a thing, because everything we do, if it's not an object, is convert it to an array. So it will then only affect the case, where we have an `object` which is not an array and will then do [obj1, obj2, ...]. Now, even if its an empty object, we can call emptyObject[0] and emptyObj.length, which will both be undefined and then skipped, same with length.. but yeah probably more convenient to use Array.isArray

Reply:
I would prefer it to have clean checks here ... but you are right